scripts/packaging/createupmpackages.ps1

```diff
@@ -169,11 +164,5 @@ foreach ($entry in $packages.GetEnumerator()) {
     Start-Process -FilePath "git" -ArgumentList "checkout package.json" -NoNewWindow -Wait
 }
 
-# Return the the scripts\packaging folder
-Set-Location -Path $scriptPath
-
-if ($nodejsInstalled -eq $false) {
-    # Cleanup the node.js "installation"
-    Remove-Item ".\$archiveName" -Recurse -Force
-    Remove-Item $archiveFile -Force
-}
+# Return to the starting path
+Set-Location $startPath

```

Comment:
Do you know if Set-Location affects the current location even outside of script crashes/exits?

i.e. if something above dies before we have a chance to call Set-Location to restore things, would we possibly hose all future scripts?

Reply:
good question. i believe i crash will potentially impact other scripts. fortunately, as part of CI, it will fail the run and terminate.

Reply:
I'll let this one be for now - you're right that this shouldn't end up having a real effect given how pipelines are currently setup.

Nevertheless, we should get rid of dependencies on current working going forward. Especially since we actually get ProjectRoot here... it seems like we could manipulate that.

Reply:
the current working folder is just to put the user back at their starting point, rather than a "random" location in their file system.

that is the sole use for this variable.

Reply:
doing this should protect any later scripts in the pipeline that presume a specific location